openedx-unsupported/ecommerce
# ENT-1281 Enterprise coupon screen frontend

**Description**
This PR introduces a new screen for creating and editing Enterprise Coupons, utilizing the endpoints created in https://github.com/edx/ecommerce/pull/1969. It is essentially a pared down version of the existing Coupons admin screen. Notable differences include the elimination of most of the course/catalog options, replaced by specifying a required enterprise customer catalog, and the client name being removed so that the linked enterprise customer's name is used as the client name. The list and detail page also include details about the enterprise and enterprise catalog and remove the fields that were not relevant. Only "migrated" enterprise coupons will show in the list screen, and will be editable and be able to be created once the waffle switch for the feature is flipped to on.

**Test Instructions**
Deployed to the business sandbox.

There should be a new tab in the admin tool page in ecommerce called Enterprise Coupons. Clicking on this will take you to the list page, showing only coupons that have been "migrated". A "migrated" enterprise coupon is one that has the attached business client model with the enterprise uuid set, as well as enterprise conditional offers for each voucher available. To migrate a coupon, go to the old coupon screen, find a coupon to edit, and save it with an attached enterprise and enterprise catalog.
The headings on the list page are name, created, status, client, enterprise, enterprise catalog. The coupon report download link should work exactly the same as the original coupon screen.

If you click into a coupon to see it's details, the page should display the same details except for catalog and course details that are no longer relevant, as well as including the enterprise fields.

If you click into the edit or create screen, you will see the same fields with the exception of the catalog/course toggle on the right, which is replaced with enterprise and enterprise catalog, both required. Other fields, including usage, dates, and invoice, should behave the same as the old screen.

_With the ENTERPRISE_OFFERS_FOR_COUPONS_SWITCH  off_
If you attempt to create or edit a coupon in the enterprise coupon screen, you will receive an error. The new screen is essentially read only.

"Migrated" enterprise coupons will still be listed in the old coupon screen, and will also be editable. You can also create coupons with enterprise data in the old screen, which will automatically be in the "migrated" state.

_With the ENTERPRISE_OFFERS_FOR_COUPONS_SWITCH  on_
On the old coupon list screen, "migrated" coupons will no longer appear. Coupons with enterprise data that have not been migrated will still be editable, but once edited, they will no longer appear in that screen. You will no longer be able to create coupons with enterprise data from the old screen.

On the new enterprise coupon list screen, all "migrated" coupons will appear. They should now be editable. New enterprise coupons can now be created from the new enterprise coupon screen, and will require an enterprise customer and enterprise customer catalog. These new coupons will not show up on the old coupon list screen.
